R/db-mssql.R

```diff
@@ -0,0 +1,152 @@
+#' @export
+`db_desc.Microsoft SQL Server` <- function(x) {
+  info <- DBI::dbGetInfo(x)
+  host <- if (info$host == "") "localhost" else info$host
+
+  paste0("Microsoft SQL Server ", info$serverVersion, " [", info$user, "@",
+         host, ":", info$port, "/", info$dbname, "]")
+}
+
+#' @export
+`sql_select.Microsoft SQL Server`<- function(con, select, from, where = NULL,
+                                             group_by = NULL, having = NULL,
+                                             order_by = NULL,
+                                             limit = NULL,
+                                             distinct = FALSE,
+                                             ...) {
+  out <- vector("list", 7)
+  names(out) <- c("select", "from", "where", "group_by",
+                  "having", "order_by","limit")
+
+  assert_that(is.character(select), length(select) > 0L)
+  out$select <- build_sql(
+    "SELECT ",
+
+    if (!is.null(limit) && !identical(limit, Inf)) {
+      assert_that(is.numeric(limit), length(limit) == 1L, limit > 0)
+      sql("TOP ")
+      sql(format(trunc(limit), scientific = FALSE))
+      sql(" ")},
+
+    if (distinct) sql("DISTINCT "),
+    escape(select, collapse = ", ", con = con)
+  )
+
+  assert_that(is.character(from), length(from) == 1L)
+  out$from <- build_sql("FROM ", from, con = con)
+
+  if (length(where) > 0L) {
+    assert_that(is.character(where))
+    where_paren <- escape(where, parens = TRUE, con = con)
+    out$where <- build_sql("WHERE ", sql_vector(where_paren, collapse = " AND "))
+  }
+
+  if (length(group_by) > 0L) {
+    assert_that(is.character(group_by))
+    out$group_by <- build_sql(
+      "GROUP BY ",
+      escape(group_by, collapse = ", ", con = con)
+    )
+  }
+
+  if (length(having) > 0L) {
+    assert_that(is.character(having))
+    out$having <- build_sql(
+      "HAVING ",
+      escape(having, collapse = ", ", con = con)
+    )
+  }
+
+  if (length(order_by) > 0L) {
+    assert_that(is.character(order_by))
+    out$order_by <- build_sql(
+      "ORDER BY ",
+      escape(order_by, collapse = ", ", con = con)
+    )
+  }
+
+  escape(unname(compact(out)), collapse = "\n", parens = FALSE, con = con)
+}
+
+
+#' @export
+`sql_subquery.Microsoft SQL Server` <- function(con, from, name = dbplyr:::unique_name(), ...) {
```

Comment:
Isn't this the same as `sql_subquery.DBIConnection`?  You don't need to override methods that are the same as for DBIConnection.